### PR TITLE
Introduce endpoints for /state/*

### PR DESCRIPTION
**Pull request by shamsimam**
Add `/state/***` endpoints for the different components. Modifies the integration tests to use these new endpoints.

### CODE REVIEW

File: waiter/src/waiter/handler.clj [hunk 1]
```diff
@@ -491,21 +491,33 @@
...
-                          state-chan ([state-data] state-data)
-                          (async/timeout timeout-ms) ([_] {:message "Query for router state timed out"})
-                          :priority true)
-          scheduler-state (let [response-chan (async/promise-chan)]
-                            (async/>!! scheduler-chan {:response-chan response-chan})
-                            (async/alt!!
-                              response-chan ([state] state)
-                              (async/timeout timeout-ms) ([_] {:message "Query for scheduler state timed out"})
-                              :priority true))]
+          current-state (async/<!! (retrieve-maintainer-state state-chan timeout-ms))
```
Comment (sradack) [inline waiter/src/waiter/handler.clj]: Fix conflicts

File: waiter/src/waiter/core.clj [hunk 2]
```diff
@@ -915,6 +921,49 @@
                                                              determine-priority-fn ws/process-response! ws/process-exception-in-request
                                                              ws/abort-request-callback-factory request))]
                                          (ws/request-handler password process-request-fn request)))))
+   :display-kv-store-state-handler-fn (pc/fnk [[:curator kv-store]
```
Comment (sradack) [inline waiter/src/waiter/core.clj]: Can we remove `display-` from all these handler names?

**Comment by shamsimam** [comment]
> We can handle fixing integration tests that use /state directly when we end up removing that functionality. 

This PR already handles some (if not all) of the calls to `/state` in the integration tests.

File: waiter/src/waiter/handler.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/handler.clj]: Done.

File: waiter/src/waiter/core.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/core.clj]: Done.

**Comment by shamsimam** [comment]
RTM

**Comment by shamsimam** [comment]
Will be updating the `/state/leader` endpoint with the leader id.

**Comment by shamsimam** [comment]
RTM

**Pull request merged**